ui/src/router/components/form/components/router_config/rule_card/RuleCard.js

```diff
@@ -150,56 +173,68 @@ export const RuleCard = ({
         }
         fullWidth>
         <Fragment>
-          {[...rule.routes, "_none_"].map((route, idx) => (
-            <Fragment key={`rule-routes-${idx}`}>
-              <EuiFlexGroup
-                className="euiFlexGroup--trafficRulesRow"
-                direction="row"
-                gutterSize="m"
-                alignItems="flexStart">
-                <EuiFlexItem grow={true} className="euiFlexItem--content">
-                  <EuiFormRow
-                    isInvalid={!!get(errors, `routes.${idx}`)}
-                    error={
-                      Array.isArray(get(errors, `routes.${idx}`))
-                        ? get(errors, `routes.${idx}`)
-                        : []
-                    }
-                    fullWidth>
-                    <EuiSuperSelect
-                      fullWidth
-                      hasDividers
-                      options={
-                        idx < rule.routes.length
-                          ? routesOptions(route)
-                          : [noneRoute, ...routesOptions(route)]
-                      }
-                      valueOfSelected={route}
-                      onChange={onChange(`routes.${idx}`)}
+        {[...rule.routes, "_none_"].map((route, idx) => {
+          // Hide last row when all routes are selected
+          if (idx < routes.length) {
+            return (
+              <Fragment key={`rule-routes-${idx}`}>
+                <EuiFlexGroup
+                  className="euiFlexGroup--trafficRulesRow"
+                  direction="row"
+                  gutterSize="m"
+                  alignItems="flexStart">
+                  <EuiFlexItem grow={true} className="euiFlexItem--content">
+                    <EuiFormRow
                       isInvalid={!!get(errors, `routes.${idx}`)}
-                    />
-                  </EuiFormRow>
-                </EuiFlexItem>
-
-                <EuiFlexItem
-                  grow={false}
-                  className="euiFlexItem--hasActions routes">
-                  {idx < rule.routes.length ? (
-                    <EuiButtonIcon
-                      size="s"
-                      color="danger"
-                      iconType="trash"
-                      onClick={onDeleteRoute(idx)}
-                      aria-label="Remove rule route"
-                    />
-                  ) : (
-                    <EuiIcon type="empty" size="l" />
-                  )}
-                </EuiFlexItem>
-              </EuiFlexGroup>
-              <EuiSpacer size="s" />
-            </Fragment>
-          ))}
+                      error={
+                        Array.isArray(get(errors, `routes.${idx}`))
```

Comment:
Is it possible it's not an array but an object or primitive value? Or is the `Array.isArray` simply meant to check that the field has an error? Can we instead just check this like `!!get(errors, `routes.${idx}`) ? ...` ?

Reply:
Good catch, I'll be removing this, because essentially for each route selection option's error, it'll always be a string like `At least one route should be attached to the rule`, so the `isArray` check is always false, i.e https://github.com/caraml-dev/turing/blob/main/ui/src/router/components/form/components/router_config/rule_card/RuleCard.js#L164 has never been working correctly 😅 

It is sufficient to have the error propagated from its parent `EuiFormRow` here - https://github.com/caraml-dev/turing/blob/main/ui/src/router/components/form/components/router_config/rule_card/RuleCard.js#L148.